app/javascript/pages/Affiliates/New.tsx

```diff
@@ -0,0 +1,276 @@
+
+    if (!email) newErrors.set("email", "Email is required");
+    else if (!isValidEmail(email)) newErrors.set("email", "Please enter a valid email address");
+
+    if (applyToAllProducts && (!feePercent || feePercent < 1 || feePercent > 90)) {
+      newErrors.set("feePercent", "Commission must be between 1% and 90%");
+    }
+
+    if (!applyToAllProducts && products.every((p) => !p.enabled)) {
+      newErrors.set("products", "Please enable at least one product");
+    }
+
+    if (!applyToAllProducts && products.some((p) => p.enabled && (!p.fee_percent || p.fee_percent < 1 || p.fee_percent > 90))) {
+      newErrors.set("products", "All enabled products must have commission between 1% and 90%");
+    }
+
+    if (destinationUrl && destinationUrl !== "" && !isUrlValid(destinationUrl)) {
+      newErrors.set("destinationUrl", "Please enter a valid URL");
+    }
+
+    if (newErrors.size > 0) {
+      setErrors(newErrors);
+      const [, firstError] = Array.from(newErrors.entries())[0] || [];
+      if (firstError) showAlert(firstError, "error");
+      return;
+    }
+
+    try {
+      setIsSubmitting(true);
+      await addAffiliate({
+        email,
+        products: products.map((p) => ({
+          id: p.id,
+          enabled: p.enabled,
+          name: p.name,
+          fee_percent: p.fee_percent,
+          destination_url: p.destination_url || null,
+          referral_url: p.referral_url,
+        })),
+        fee_percent: feePercent,
+        apply_to_all_products: applyToAllProducts,
+        destination_url: destinationUrl,
+      });
+      showAlert("Affiliate added successfully!", "success");
+      router.visit(Routes.affiliates_path());
+    } catch (e) {
+      assertResponseError(e);
+      showAlert((e as Error).message || "Failed to add affiliate", "error");
+    } finally {
+      setIsSubmitting(false);
+    }
+  });
+
+  return (
+    <div>
+      <PageHeader
+        className="sticky-top"
+        title="New Affiliate"
+        actions={
+          <>
+            <NavigationButtonInertia href={Routes.affiliates_path()} disabled={isSubmitting}>
+              <Icon name="x-square" />
+              Cancel
+            </NavigationButtonInertia>
+            <Button
+              color="accent"
+              onClick={handleSubmit}
+              disabled={isSubmitting || !loggedInUser?.policies.direct_affiliate.create}
+            >
+              {isSubmitting ? "Adding..." : "Add affiliate"}
+            </Button>
+          </>
+        }
+      />
+      <form>
+        <section className="p-4! md:p-8!">
+          <header
+            dangerouslySetInnerHTML={{
+              __html:
+                "Add a new affiliate below and we'll send them a unique link to share with their audience. Your affiliate will then earn a commission on each sale they refer. <a href='/help/article/333-affiliates-on-gumroad' target='_blank' rel='noreferrer'>Learn more</a>",
+            }}
```

Comment:
I know you're merely moving code here. I wonder if it is really necessary to set inner HTML here.